mkosi

```diff
@@ -1600,56 +1595,63 @@ def invoke_tdnf(args: CommandLineArguments,
     with mount_api_vfs(args, root):
         run(cmdline, check=True)
 
-@completestep('Installing Photon')
-def install_photon(args: CommandLineArguments, root: str, do_run_build_script: bool) -> None:
-    masked = disable_kernel_install(args, root)
-
-    gpg_key = '/etc/pki/rpm-gpg/VMWARE-RPM-GPG-KEY'
-    gpg_key_string = f'file://{gpg_key}'
+class Repo(NamedTuple):
+    id: str
+    name: str
+    url: str
+    gpgpath: str
+    gpgurl: Optional[str] = None
 
-    if os.path.exists(gpg_key):
-        gpgcheck = "gpgcheck=1"
-        cmdline = ["rpm", "--import", gpg_key, "--root", root]
-        run(cmdline, check=True)
+def setup_dnf(args: CommandLineArguments, root: str, repos: List[Repo] = []) -> None:
+    gpgcheck = True
 
-    else:
-        gpgcheck = "gpgcheck=0"
+    repo_file = os.path.join(workspace(root), "temp.repo")
+    with open(repo_file, 'w') as f:
+        for (id, name, url, gpgpath, gpgurl) in repos:
```

Comment:
It might make sense not to destructure this here, since named tuples allow dot-access, but I don't particularly mind either way. Whatever you like better.